src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1543,6 +1562,90 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
             return BetterResult.Neither;
         }
 
+        private static BetterResult BetterParamArrayTypeTieBreaker(TypeSymbol type1, TypeSymbol type2)
+        {
+            // Duplicating VS2013 logic, relevant code is in 
+            // BetterTypeEnum BSYMMGR::CompareTypes(TypeArray * ta1, TypeArray * ta2)
+
+            if (type1 == type2)
+            {
+                return BetterResult.Neither;
+            }
+
+            if (type1.IsTypeParameter())
+            {
+                if (type2.IsTypeParameter())
+                {
+                    return BetterResult.Neither;
+                }
+
+                return BetterResult.Right;
+            }
+            else if (type2.IsTypeParameter())
+            {
+                return BetterResult.Left;
+            }
+
+            if (type1.Kind == type2.Kind)
+            {
+                switch (type1.Kind)
+                {
+                    case SymbolKind.NamedType:
+                    case SymbolKind.ErrorType:
+
+                        var t1 = (NamedTypeSymbol)type1;
+                        var t2 = (NamedTypeSymbol)type2;
+
+                        var builder1 = ArrayBuilder<TypeSymbol>.GetInstance();
+                        var builder2 = ArrayBuilder<TypeSymbol>.GetInstance();
+                        HashSet<DiagnosticInfo> h = null;
+
+                        t1.GetAllTypeArguments(builder1, ref h);
+                        t2.GetAllTypeArguments(builder2, ref h);
+
+                        // The one with more arguments is more specific.
+                        if (builder1.Count > builder2.Count)
+                        {
+                            return BetterResult.Left;
+                        }
+                        else if (builder1.Count < builder2.Count)
+                        {
+                            return BetterResult.Right;
+                        }
+
+                        BetterResult result = BetterResult.Neither;
+
+                        for (int i = 0; i < builder1.Count; i++)
```

Comment:
I understand what this is doing but I can't think of any rational explanation for why it should be this way.